src/scripting/lua_attributes.hpp

```diff
@@ -0,0 +1,177 @@
+/*
+	Copyright (C) 2009 - 2024
+	Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY.
+
+	See the COPYING file for more details.
+*/
+
+/// Attribute registration system, mainly for objects with a lot of attributes, like units
+/// Not used for GUI2 widgets, as they're even more complicated with a deep hierarchy.
+
+#pragma once
+
+struct lua_State;
+class t_string;
+class vconfig;
+
+#include "config.hpp"
+#include "variable_info.hpp"
+#include "map/location.hpp"
+
+#include <string>
+#include <string_view>
+#include <vector>
+
+template<typename T>
+std::decay_t<T> lua_check(lua_State *L, int n);
+
+/// Holds a lookup table for members of one type of object.
+struct luaW_Registry {
+	inline static std::map<std::string_view /* metatable */, std::reference_wrapper<luaW_Registry>> lookup;
```

Comment:
What owns the strings that the string_views point to? If it's private_metatable or public_metatable, those are both destroyed before this member.

Reply:
They are most likely status string literals.

Reply:
Actually I might be wrong I didn't really read the code

Reply:
They are static string literals. Those literals _are_ copied into `private_metatable` or `public_metatable`, but the string_view points to the original literal.